Ray.jl/test/utils.jl

```diff
@@ -21,3 +21,58 @@ function setup_core_worker(body)
         rayjll.shutdown_driver()
     end
 end
+
+# Useful in running tests which require us to re-run `Ray.init` which currently can only be
+# called once as it is infeasible to reset global `Ref`'s using `isassigned` conditionals.
+macro process_eval(ex)
+    expr = QuoteNode(ex)
+    return esc(:(process_eval($expr)))
+end
```

Comment:
could mock `init` too?

Reply:
I considered using Mocking for this but realized quickly for the tests which validate we don't accidentally load unnecessary packages by default would be extremely hard to get right and went with this approach instead.

Reply:
ah yeah taht's a good point